munkipkg

```diff
@@ -619,6 +659,92 @@ def build_distribution_pkg(build_info, options):
         raise BuildError(err)
 
 
+def notarize(build_info, options):
+    '''Use xcrun altool to notarize our package'''
+    if not (
+        'username' in build_info['notarization_info']
+        and 'password' in build_info['notarization_info']
+    ):
+        raise MunkiPkgError("notarization_info lacks username or password.")
+
+    display("Uploading package to Apple notary service", options.quiet)
+    cmd = [
+        XCRUN,
+        'altool',
+        '--notarize-app',
+        '--primary-bundle-id',
+        build_info['identifier'],
+        '--username',
+        build_info['notarization_info']['username'],
+        '--password',
+        build_info['notarization_info']['password'],
+        '--output-format',
+        'xml',
+        '--file',
+        os.path.join(build_info['build_dir'], build_info['name']),
+    ]
+    if 'asc_provider' in build_info['notarization_info']:
+        cmd.extend(
+            ['--asc-provider', build_info['notarization_info']['asc_provider']]
+        )
+    retcode, proc_stdout, proc_stderr = run_subprocess(cmd)
+    try:
+        output = readPlistFromString(proc_stdout)
+    except ExpatError:
+        print(proc_stderr, file=sys.stderr)
+        raise MunkiPkgError("Notarization failed. Unable to run xcrun altool")
+
+    if retcode:
+        for product_error in output['product-errors']:
+            print("altool: FAILURE " + product_error['message'], file=sys.stderr)
+        raise MunkiPkgError("Notarization failed")
+    else:
+        display(
+            "RequestUUID " + output['notarization-upload']['RequestUUID'],
+            options.quiet,
+            "altool",
+        )
+        display("SUCCESS " + output['success-message'], options.quiet, "altool")
+
+
+def staple(build_info, options):
+    '''Use xcrun staple to add staple to our package'''
+    display("Stapling package", options.quiet)
+    cmd = [
+        XCRUN,
+        'stapler',
+        'staple',
+        os.path.join(build_info['build_dir'], build_info['name']),
+    ]
+
+    if 'stapler_timeout' in build_info['notarization_info']:
+        timeout = build_info['notarization_info']['stapler_timeout']
+    else:
+        timeout = STAPLER_TIMEOUT
```

Comment:
These four lines are more concisely written:
`timeout = build_info['notarization_info'].get('stapler_timeout', STAPLER_TIMEOUT)`

Reply:
Implemented in ee6520c.